# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/OriginInfo.cpp
# Commit: 9987ebf11f14
# Full Hash: 9987ebf11f141d7e31c958972aee01676d717246
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2025-06-16 09:38:11
# Regressor Bug: 1962133
# File Overlap Count: 14
# Description:
#   Bug 1962133 - QM: Add accessed flag to FullOriginMetadata and persist it in the origin metadata file; r=dom-storage-reviewers,hsingh
#   
#   This patch introduces a new mAccessed field in OriginStateMetadata, which is
#   inherited by FullOriginMetadata. One of the reserved 32-bit fields in the
#   metadata header is now used as a bitfield, with one bit allocated to represent
# ==============================================================================

diff -r d2ec302b7103 -r 9987ebf11f14 dom/quota/OriginInfo.cpp
--- a/dom/quota/OriginInfo.cpp	Sun Jun 15 21:21:24 2025 +0000
+++ b/dom/quota/OriginInfo.cpp	Sun Jun 15 23:24:14 2025 +0000
@@ -70,7 +70,7 @@
 OriginStateMetadata OriginInfo::LockedFlattenToOriginStateMetadata() const {
   AssertCurrentThreadOwnsQuotaMutex();
 
-  return {mAccessTime, mPersisted};
+  return {mAccessTime, mAccessed, mPersisted};
 }
 
 FullOriginMetadata OriginInfo::LockedFlattenToFullOriginMetadata() const {